src/index/base.cpp

```diff
@@ -66,6 +66,29 @@ bool BaseIndex::Init()
         m_best_block_index = FindForkInGlobalIndex(chainActive, locator);
     }
     m_synced = m_best_block_index.load() == chainActive.Tip();
+
+    if (!m_synced) {
+        bool prune_violation = false;
+        if (!m_best_block_index) {
+            // it looks like this index is not built yet
+            // make sure we have all data back to the genesis
+            const CBlockIndex* block = chainActive.Tip();
+            while (block->pprev && (block->pprev->nStatus & BLOCK_HAVE_DATA)) {
```

Comment:
0d8f7369cd8b51dfb7b3b9b920e4e11f7962faa9 

This doesn't check `status & BLOCK_HAVE_DATA` on the tip, is that intentional?

Reply:
I don't think the tip can ever not have data?

Reply:
> I don't think the tip can ever not have data? 

Jup.